pkg/util/k8s/k8s.go

```diff
@@ -4,45 +4,23 @@ package k8s
 import (
 	"context"
 	"fmt"
-	"os"
-	"sort"
-	"time"
+	"log"
 
-	docker_types "github.com/docker/docker/api/types"
-	docker_nat "github.com/docker/go-connections/nat"
 	gragent "github.com/grafana/agent/pkg/operator/apis/monitoring/v1alpha1"
-	k3d_client "github.com/k3d-io/k3d/v5/pkg/client"
-	config "github.com/k3d-io/k3d/v5/pkg/config"
-	k3d_cfgtypes "github.com/k3d-io/k3d/v5/pkg/config/types"
-	k3d_config "github.com/k3d-io/k3d/v5/pkg/config/v1alpha4"
-	k3d_log "github.com/k3d-io/k3d/v5/pkg/logger"
-	k3d_runtime "github.com/k3d-io/k3d/v5/pkg/runtimes"
-	k3d_docker "github.com/k3d-io/k3d/v5/pkg/runtimes/docker"
-	k3d_types "github.com/k3d-io/k3d/v5/pkg/types"
-	k3d_version "github.com/k3d-io/k3d/v5/version"
 	promop_v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/testcontainers/testcontainers-go/modules/k3s"
 	apiextensions_v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/rand"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
-	k8s_clientcmd "k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // Cluster is a Kubernetes cluster that runs inside of a k3s Docker container.
 // Call GetConfig to retrieve a Kubernetes *rest.Config to use to connect to
 // the cluster.
 //
-// Cluster also runs an NGINX ingress controller which is exposed to the host.
-// Call GetHTTPAddr to get the address for making requests against the server.
-//
-// Set K8S_USE_DOCKER_NETWORK in your environment variables if you are
-// running tests from inside of a Docker container. This environment variable
-// configures the k3s Docker container to join the same network as the
-// container tests are running in. When this environment variable isn't set,
-// the exposed ports on the Docker host are used for cluster communication.
```

Comment:
This is no longer required because testcontainers-go has logic to figure out if it's running in Docker automatically.